WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/Onboarding Errors/InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModel.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import Yosemite
+
+class InPersonPaymentsCashOnDeliveryPaymentGatewayNotSetUpViewModel: ObservableObject {
+    let completion: () -> ()
+    private let stores: StoresManager
+    private let noticePresenter: NoticePresenter
+
+    @Published var awaitingResponse = false
+
+    private var siteID: Int64? {
+        stores.sessionManager.defaultStoreID
+    }
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         noticePresenter: NoticePresenter = ServiceLocator.noticePresenter,
+         completion: @escaping () -> ()) {
+        self.stores = stores
+        self.noticePresenter = noticePresenter
+        self.completion = completion
+    }
+
+    func skipTapped() {
+        guard let siteID = siteID else {
+            return completion()
+        }
+
+        let action = AppSettingsAction.setSkippedCashOnDeliveryOnboardingStep(siteID: siteID)
+        stores.dispatch(action)
+        completion()
+    }
+
+    func enableTapped() {
+        guard let siteID = siteID else {
+            return completion()
+        }
+
+        awaitingResponse = true
+
+        let action = PaymentGatewayAction.updatePaymentGateway(defaultCashOnDeliveryGateway(siteID: siteID)) { [weak self] result in
+            guard let self = self else { return }
+            guard result.isSuccess else {
+                DDLogError("💰 Could not update Payment Gateway: \(String(describing: result.failure))")
+                self.awaitingResponse = false
+                self.displayEnableCashOnDeliveryFailureNotice()
+                return
+            }
+
+            self.completion()
```

Comment:
Do we have to set `awaitingResponse` to `false` when it completes successfully?

Reply:
Good spot, but no, I think it looks better when we don't do that. 

I tried it originally, and it results in the button flickering back to the enabled state as the view is pushed off screen, which is a bit jarring.